pyparsing/core.py

```diff
@@ -2829,78 +2837,81 @@ def __init__(
+            else:
+                repeat = f"{{{self.minLen}}}"
+            self.reString = f"{re_leading_fragment}{repeat}"
+        else:
+            if max == 1:
+                re_body_fragment = ""
+                repeat = ""
             else:
-                if max == 1:
-                    re_body_fragment = ""
-                    repeat = ""
+                re_body_fragment = f"[{_collapse_string_to_ranges(self.bodyChars)}]"
+                if max == 0 and self.minLen == 1:
+                    repeat = "*"
+                elif max == 2:
+                    repeat = "?" if min <= 1 else ""
+                elif min != max:
+                    repeat = f"{{{min - 1 if min > 0 else ''},{max - 1 if max > 0 else ''}}}"
                 else:
-                    re_body_fragment = f"[{_collapse_string_to_ranges(self.bodyChars)}]"
-                    if max == 0 and self.minLen == 1:
-                        repeat = "*"
-                    elif max == 2:
-                        repeat = "?" if min <= 1 else ""
-                    else:
-                        if min != max:
-                            repeat = f"{{{min - 1 if min > 0 else ''},{max - 1 if max > 0 else ''}}}"
-                        else:
-                            repeat = f"{{{min - 1 if min > 0 else ''}}}"
+                    repeat = f"{{{min - 1 if min > 0 else ''}}}"
 
-                self.reString = f"{re_leading_fragment}{re_body_fragment}{repeat}"
+            self.reString = f"{re_leading_fragment}{re_body_fragment}{repeat}"
 
-            if self.asKeyword:
-                self.reString = rf"\b{self.reString}\b"
+        if self.asKeyword:
+            self.reString = rf"\b{self.reString}\b"
 
-            try:
-                self.re = re.compile(self.reString)
-            except re.error:
-                self.re = None  # type: ignore[assignment]
-            else:
-                self.re_match = self.re.match
-                self.parseImpl = self.parseImpl_regex  # type: ignore[assignment]
+        try:
+            self.re = re.compile(self.reString)
+        except re.error:
+            self.re = None  # type: ignore[assignment]
+        else:
+            self.re_match = self.re.match
+            self.parseImpl = self.parseImpl_regex  # type: ignore[assignment]
 
     def _generateDefaultName(self) -> str:
         def charsAsStr(s):
             max_repr_len = 16
             s = _collapse_string_to_ranges(s, re_escape=False)
+
             if len(s) > max_repr_len:
                 return s[: max_repr_len - 3] + "..."
-            else:
-                return s
+
+            return s
 
         if self.initChars != self.bodyChars:
             base = f"W:({charsAsStr(self.initChars)}, {charsAsStr(self.bodyChars)})"
         else:
             base = f"W:({charsAsStr(self.initChars)})"
 
         # add length specification
-        if self.minLen > 1 or self.maxLen != _MAX_INT:
```

Comment:
This logic changes needs to be broken out in a separate PR (and possibly needs additional tests).